common/data_refinery_common/models/models.py

```diff
@@ -294,6 +294,30 @@ def save(self, *args, **kwargs):
         self.last_modified = current_time
         return super(ExperimentAnnotation, self).save(*args, **kwargs)
 
+class Pipeline(models.Model):
+    "Pipeline that is associated with a series of ComputationalResult records."""
+
+    name = models.CharField(max_length=255)
```

Comment:
After review the code, I'm thinking that this should maybe be a ChoiceField or at least use an Enum or something somwhere so we don't have the problem of having to know `No Op` versus `noop` versus `NO_OP` vs etc. etc. etc. 

This goes for everything

Reply:
I can definitely see the advantages of using Enum or ChoiceField. There is one disadvantage though: Every time when we add a new pipeline, we will have to modify the model and migrate the database.

Reply:
Well we don't necessarily need to have it be at the database level. I think we need it somewhere, it could be an Enum class or even just a namespace full of `SALMON_QUANT = "SALMON_QUANT"` so that you could do `PipelineEnums.SALMON_QUANT` instead of hardcoding the string. I think this might help looking up what the possible values are along with keeping them consistent. We have at least the following values:
`"Salmon Quant", "Tximport", "MultiQC", "Salmontools", "Submitter-processed", "Agilent SCAN TwoColor"` which aren't very consistent case-wise.

I think I would prefer a naming convention like `ALL_CAPS_v1.2.3`. 

However I was also thinking that it might make sense to actually have the version be a separate field so that if we ever wanted to get all the pipelines for `AGILENT_SCAN_TWOCOLOR` we could just query with `WHERE name='AGILENT_SCAN_TWOCOLOR' rather than `WHERE name LIKE 'AGILENT_SCAN_TWOCOLOR%'`. Also since we're using the version of the system for the version number we could query for all results from a given version.

Reply:
In the new database model, the terms such as "Salmon Quant", "Tximport", "MultiQC", "Salmontools", "Submitter-processed" and "Agilent SCAN TwoColor" are actually names of **processors**, not names of **pipelines**. 

The possible values of pipelines are hardcoded at the end of each pipeline module (`salmon.py`, `agilent_twocolor.py`, etc)  right before the call of `utils.run_pipeline()`. 

The naming convention of `ALL_CAPS_v1.2.3` is a good point. But I'm not sure whether these processor names will be shown on web UIs. If they will, then a name with white spaces seems to be more friendly to the front end because it can be displayed directly.  

I agree that having the processor and pipelines hardcoded everywhere is not a good idea. Let me put them at the beginning of each pipeline module.

Having a separate `version` field is a good idea too. I will add it in the model. Thanks.

Reply:
Ah yeah that's a good point about user-friendly names.